### PR TITLE
Country key package download and risk detection (INTEROP 1660-2510)

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
This PR makes use of the clients country API to process country packages.

## Checklist

* [x] This PR does not change text that hasn't been signed off with the RKI. Have a look at the pinned issue [440](https://github.com/corona-warn-app/cwa-app-ios/issues)
* [ ] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [ ] If this PR comes from a fork, please [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
* [x] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes # 41)
* [x] [Link your Pull Request to an issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) - Pull Requests that are not linked to an issue with you as an assignee will be closed.
* [ ] Create Work In Progress [WIP] pull requests only if you need clarification or an explicit review before you can continue your work item.
* [x] Make sure that your PR is not introducing _unnecessary_ reformatting (e.g., introduced by on-save hooks in your IDE)

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Client/Client.swift [hunk 1]
```diff
@@ -102,8 +112,23 @@ protocol Client {
...
 		completion: @escaping HourCompletionHandler
 	)
+
+	// MARK: Getting the Configuration
+
+	typealias ExposureConfigurationCompletionHandler = (ENExposureConfiguration?) -> Void
+
+	/// Gets the remove exposure configuration. See `ENExposureConfiguration` for more details
+	/// Parameters:
+	/// - completion: Will be called with the remove configuration or an error if something went wrong. The completion handler will always be called on the main thread.
```
Comment (ChristianKienle) [inline src/xcode/ENA/ENA/Source/Client/Client.swift:122]: ♥️

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+ExposureConfigTests.swift [hunk 2]
```diff
@@ -26,6 +26,98 @@ final class HTTPClientExposureConfigTests: XCTestCase {
...
 
+	#if INTEROP
+
+	func testInvalidEmptyExposureConfigurationResponseData() {
+		let stack = MockNetworkStack(
+			httpStatus: 200,
+			responseData: nil
+		)
+
+		let expectation = self.expectation(description: "HTTPClient should have failed.")
```
Comment (ChristianKienle) [inline src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+ExposureConfigTests.swift:37]: For future tests you could consider setting [`assertforoverfulfill`](https://developer.apple.com/documentation/xctest/xctestexpectation/2806575-assertforoverfulfill) to true like this:

```swift
expectation.assertforoverfulfill = true
```

Then calling `fulfill` too frequently will cause the test to fail. I personally have also not done that consistently but I think this is a good habit to pick up. (Note to myself). :D

File: src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetectionExecutor.swift [hunk 3]
```diff
@@ -52,7 +54,144 @@ final class ExposureDetectionExecutor: ExposureDetectionDelegate {
...
+
+		let fileManager = FileManager()
+		let rootDir = fileManager.temporaryDirectory.appendingPathComponent(UUID().uuidString, isDirectory: true)
+		do {
+			try fileManager.createDirectory(at: rootDir, withIntermediateDirectories: true, attributes: nil)
+
+			let writer = AppleFilesWriter(rootDir: rootDir)
+
+			if store.hourlyFetchingEnabled {
+				let hourlyPackages = downloadedPackagesStore.hourlyPackages(for: .formattedToday(), country: country)
```
Comment (ChristianKienle) [inline src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetectionExecutor.swift:138]: Not sure if the following aspect is handled somewhere else but to us `hourlyFetchingEnabled` means that we only use the **three** most recent hours - not all hours from today. I was unable to find this logic in our current *develop* but it was there up until recently. Do you know what happened there? :) 

Screen Shot 2020-09-11 at 18 26 09

Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetectionExecutor.swift:138]: Good one. I think it got lost on the way. Willl add it again

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+ExposureConfigTests.swift (hunk 2 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+ExposureConfigTests.swift:37]: I think its true by default when you create it with self.expectation(...)

File: src/xcode/ENA/ENA/Source/Client/Client.swift (hunk 1 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Client/Client.swift:122]: Thanks! Need to handover this ❤️ to the creator of this comment. (I just copied)

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+ExposureConfigTests.swift (hunk 2 above)
Comment (ChristianKienle) [reply src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+ExposureConfigTests.swift:37]: Oh right.

**Review by ChristianKienle — APPROVED** [review]

**Review by haosap — COMMENTED** [review]
Only have suggestion from naming perspective.

File: src/xcode/ENA/ENA/Source/Services/Exposure Transaction/CountryKeypackageDownloader.swift [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+import ExposureNotification
+
+#if INTEROP
+
+protocol CountryKeypackageDownloaderProtocol {
```
[the remark this answers is not in the document]
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Services/Exposure Transaction/CountryKeypackageDownloader.swift]: Good sugestion. Renamed.

**Review by haosap — APPROVED** [review]
Thanks, and nice work. 💯

**Pull request merged**